{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Concatenating all tools and strains at threshold 0, 10, 100, 1000, and 10K"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import allel\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "strain=['A_J','AKR_J','BALB_CJ','CBA_J','C3H_HeJ','DBA_2J','LP_J']\n",
    "\n",
    "tools=['BioGraph*',\n",
    "'breakdancer',\n",
    "'clever',\n",
    "'delly',\n",
    "'gasv',\n",
    "'gridss',\n",
    "'indelminer',\n",
<<<<<<< HEAD
    "#'lumpexpress',\n",
=======
    "'lumpexpress',\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "'mistrvar',\n",
    "'pindel',\n",
    "'popdel',\n",
    "'rdxplorer',\n",
    "'smoove',\n",
    "'sniffles']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 0 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
<<<<<<< HEAD
      "/home/vs/.local/lib/python3.5/site-packages/allel/io/vcf_read.py:1745: UserWarning: multiple INFO headers for 'SVLEN'\n",
      "  warnings.warn('multiple INFO headers for %r' % k)\n",
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:14: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \n"
=======
      "/Users/angelalu/opt/anaconda3/lib/python3.7/site-packages/allel/io/vcf_read.py:1745: UserWarning: multiple INFO headers for 'SVLEN'\n",
      "  warnings.warn('multiple INFO headers for %r' % k)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/0t/deletions/nf_0t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>1725</td>\n",
       "      <td>6055203</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>78</td>\n",
       "      <td>17072687</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>34582728</td>\n",
       "      <td>17098523</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>48</td>\n",
       "      <td>17201451</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>114</td>\n",
       "      <td>17231867</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>78</td>\n",
       "      <td>FP</td>\n",
       "      <td>17072687</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length  position strain threshold      tool\n",
       "0   FP      1725   6055203   LP_J         0  sniffles\n",
       "1   FP        78  17072687   LP_J         0  sniffles\n",
       "2   FP  34582728  17098523   LP_J         0  sniffles\n",
       "3   FP        48  17201451   LP_J         0  sniffles\n",
       "4   FP       114  17231867   LP_J         0  sniffles"
=======
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203         0\n",
       "1  sniffles   LP_J        78   FP  17072687         0\n",
       "2  sniffles   LP_J  34582728   FP  17098523         0\n",
       "3  sniffles   LP_J        48   FP  17201451         0\n",
       "4  sniffles   LP_J       114   FP  17231867         0"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#0 threshold deletions\n",
    "\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
<<<<<<< HEAD
    "\n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_0 = pd.concat([df_current, df_0],ignore_index=True)\n",
    "            \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
=======
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0 = pd.concat([df_current, df_0],ignore_index=True)   \n",
    "        \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "            \n",
    "df_0.head()"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n"
=======
   "execution_count": 3,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/0t/non-deletions/nf_0t.pindel.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n",
      "----\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>TN</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6055203</td>\n",
       "      <td>TN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length position strain threshold      tool\n",
       "0   TN   6055203        0   LP_J         0  sniffles\n",
       "1   TN  11015759  6056928   LP_J         0  sniffles\n",
       "2   TN  11015759  6056928   LP_J         0  sniffles\n",
       "3   TN  11015759  6056928   LP_J         0  sniffles\n",
       "4   TN  11015759  6056928   LP_J         0  sniffles"
=======
       "       tool strain    length flag position threshold\n",
       "0  sniffles   LP_J   6055203   TN        0         0\n",
       "1  sniffles   LP_J  11015759   TN  6056928         0\n",
       "2  sniffles   LP_J  11015759   TN  6056928         0\n",
       "3  sniffles   LP_J  11015759   TN  6056928         0\n",
       "4  sniffles   LP_J  11015759   TN  6056928         0"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#0 threshold non-deletions\n",
    "import os\n",
    "\n",
    "th=0\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_0_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        \n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
    "        \n",
<<<<<<< HEAD
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_0_nondel = pd.concat([df_current, df_0_nondel],ignore_index=True)\n",
    "            \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "              \n",
=======
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            print (\"----\")\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_0_nondel = pd.concat([df_current, df_0_nondel],ignore_index=True)   \n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "        \n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "df_0_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
<<<<<<< HEAD
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.A_J.chr19.100p_sorted.modified.vcf\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:18: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.A_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.BioGraph*.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.breakdancer.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.clever.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.delly.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gasv.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.gridss.LP_J.chr19.100p_sorted.modified.vcf\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.indelminer.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.mistrvar.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.popdel.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.rdxplorer.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.smoove.LP_J.chr19.100p_sorted.modified.vcf\n",
      "../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.sniffles.LP_J.chr19.100p_sorted.modified.vcf\n"
=======
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/10t/deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>1725</td>\n",
       "      <td>6055203</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>34582728</td>\n",
       "      <td>17098523</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>48</td>\n",
       "      <td>17201451</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>114</td>\n",
       "      <td>17231867</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>35</td>\n",
       "      <td>17291187</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>35</td>\n",
       "      <td>FP</td>\n",
       "      <td>17291187</td>\n",
       "      <td>10</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length  position strain threshold      tool\n",
       "0   FP      1725   6055203   LP_J        10  sniffles\n",
       "1   FP  34582728  17098523   LP_J        10  sniffles\n",
       "2   FP        48  17201451   LP_J        10  sniffles\n",
       "3   FP       114  17231867   LP_J        10  sniffles\n",
       "4   FP        35  17291187   LP_J        10  sniffles"
=======
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203        10\n",
       "1  sniffles   LP_J  34582728   FP  17098523        10\n",
       "2  sniffles   LP_J        48   FP  17201451        10\n",
       "3  sniffles   LP_J       114   FP  17231867        10\n",
       "4  sniffles   LP_J        35   FP  17291187        10"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "#tools were not able to run on mouse data; ignore?\n",
    "\n",
    "th=10     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
<<<<<<< HEAD
    "            print(file)\n",
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_10 = pd.concat([df_current, df_10],ignore_index=True)\n",
    "\n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "            \n",
    "df_10.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
<<<<<<< HEAD
   "metadata": {},
   "outputs": [],
   "source": [
    "df_10.to_csv('../analysis_files/df_10temp.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
<<<<<<< HEAD
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:15: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  from ipykernel import kernelapp as app\n"
=======
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10t/non-deletions/nf_10t.pindel.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>TN</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6055203</td>\n",
       "      <td>TN</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length position strain threshold      tool\n",
       "0   TN   6055203        0   LP_J        10  sniffles\n",
       "1   TN  11015759  6056928   LP_J        10  sniffles\n",
       "2   TN  11015759  6056928   LP_J        10  sniffles\n",
       "3   TN  11015759  6056928   LP_J        10  sniffles\n",
       "4   TN  11015759  6056928   LP_J        10  sniffles"
      ]
     },
     "execution_count": 6,
=======
       "       tool strain    length flag position threshold\n",
       "0  sniffles   LP_J   6055203   TN        0        10\n",
       "1  sniffles   LP_J  11015759   TN  6056928        10\n",
       "2  sniffles   LP_J  11015759   TN  6056928        10\n",
       "3  sniffles   LP_J  11015759   TN  6056928        10\n",
       "4  sniffles   LP_J  11015759   TN  6056928        10"
      ]
     },
     "execution_count": 5,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10 threshold non-deletions\n",
    "th=10     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
<<<<<<< HEAD
    "\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_10_nondel = pd.concat([df_current, df_10_nondel],ignore_index=True)\n",
    "\n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
=======
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10_nondel = pd.concat([df_current, df_10_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "df_10_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 100 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n"
=======
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/100t/deletions/nf_100t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>1725</td>\n",
       "      <td>6055203</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>34582728</td>\n",
       "      <td>17098523</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>48</td>\n",
       "      <td>17201451</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>35</td>\n",
       "      <td>17291187</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>34</td>\n",
       "      <td>17805736</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>35</td>\n",
       "      <td>FP</td>\n",
       "      <td>17291187</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34</td>\n",
       "      <td>FP</td>\n",
       "      <td>17805736</td>\n",
       "      <td>100</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length  position strain threshold      tool\n",
       "0   FP      1725   6055203   LP_J       100  sniffles\n",
       "1   FP  34582728  17098523   LP_J       100  sniffles\n",
       "2   FP        48  17201451   LP_J       100  sniffles\n",
       "3   FP        35  17291187   LP_J       100  sniffles\n",
       "4   FP        34  17805736   LP_J       100  sniffles"
      ]
     },
     "execution_count": 7,
=======
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203       100\n",
       "1  sniffles   LP_J  34582728   FP  17098523       100\n",
       "2  sniffles   LP_J        48   FP  17201451       100\n",
       "3  sniffles   LP_J        35   FP  17291187       100\n",
       "4  sniffles   LP_J        34   FP  17805736       100"
      ]
     },
     "execution_count": 6,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#100 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=100\n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_100 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_100 = pd.concat([df_current, df_100],ignore_index=True)\n",
    "        \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "            \n",
    "df_100.head()"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:14: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \n"
=======
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/100t/non-deletions/nf_100t.pindel.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>TN</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>100</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6055203</td>\n",
       "      <td>TN</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>100</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length position strain threshold      tool\n",
       "0   TN   6055203        0   LP_J       100  sniffles\n",
       "1   TN  11015759  6056928   LP_J       100  sniffles\n",
       "2   TN  11015759  6056928   LP_J       100  sniffles\n",
       "3   TN  11015759  6056928   LP_J       100  sniffles\n",
       "4   TN  11015759  6056928   LP_J       100  sniffles"
      ]
     },
     "execution_count": 8,
=======
       "       tool strain    length flag position threshold\n",
       "0  sniffles   LP_J   6055203   TN        0       100\n",
       "1  sniffles   LP_J  11015759   TN  6056928       100\n",
       "2  sniffles   LP_J  11015759   TN  6056928       100\n",
       "3  sniffles   LP_J  11015759   TN  6056928       100\n",
       "4  sniffles   LP_J  11015759   TN  6056928       100"
      ]
     },
     "execution_count": 7,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#100 threshold non-deletions\n",
    "\n",
    "th=100     \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_100_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
<<<<<<< HEAD
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_100_nondel = pd.concat([df_current, df_100_nondel],ignore_index=True)\n",
    "        \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)           \n",
=======
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_100_nondel = pd.concat([df_current, df_100_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "df_100_nondel.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1000 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:17: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
=======
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.A_J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.AKR_J.chr19.100p_sorted.modified.vcf\n",
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.BALB_CJ.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.CBA_J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.C3H_HeJ.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.DBA_2J.chr19.100p_sorted.modified.vcf\n",
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/1000t/deletions/nf_1000t.sniffles.LP_J.chr19.100p_sorted.modified.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>34582728</td>\n",
       "      <td>17098523</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>48</td>\n",
       "      <td>17201451</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>35</td>\n",
       "      <td>17291187</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>34</td>\n",
       "      <td>17805736</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>37</td>\n",
       "      <td>18086146</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>35</td>\n",
       "      <td>FP</td>\n",
       "      <td>17291187</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34</td>\n",
       "      <td>FP</td>\n",
       "      <td>17805736</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>37</td>\n",
       "      <td>FP</td>\n",
       "      <td>18086146</td>\n",
       "      <td>1000</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length  position strain threshold      tool\n",
       "0   FP  34582728  17098523   LP_J      1000  sniffles\n",
       "1   FP        48  17201451   LP_J      1000  sniffles\n",
       "2   FP        35  17291187   LP_J      1000  sniffles\n",
       "3   FP        34  17805736   LP_J      1000  sniffles\n",
       "4   FP        37  18086146   LP_J      1000  sniffles"
      ]
     },
     "execution_count": 9,
=======
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J  34582728   FP  17098523      1000\n",
       "1  sniffles   LP_J        48   FP  17201451      1000\n",
       "2  sniffles   LP_J        35   FP  17291187      1000\n",
       "3  sniffles   LP_J        34   FP  17805736      1000\n",
       "4  sniffles   LP_J        37   FP  18086146      1000"
      ]
     },
     "execution_count": 8,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#1000 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=1000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_1000 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
<<<<<<< HEAD
    "           # print(file)\n",
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_1000 = pd.concat([df_current, df_1000],ignore_index=True)\n",
    "            \n",
<<<<<<< HEAD
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "            \n",
    "        else:\n",
    "            print (\"Nonexistant File: \" + file)\n",
=======
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "            \n",
    "    else:\n",
    "        print (\"Nonexistant File: \" + file)\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "           \n",
    "df_1000.head()\n"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:14: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \n"
=======
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/1000t/non-deletions/nf_1000t.pindel.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>TN</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1000</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6055203</td>\n",
       "      <td>TN</td>\n",
       "      <td>0</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>1000</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length position strain threshold      tool\n",
       "0   TN   6055203        0   LP_J      1000  sniffles\n",
       "1   TN  11015759  6056928   LP_J      1000  sniffles\n",
       "2   TN  11015759  6056928   LP_J      1000  sniffles\n",
       "3   TN  11015759  6056928   LP_J      1000  sniffles\n",
       "4   TN  11015759  6056928   LP_J      1000  sniffles"
      ]
     },
     "execution_count": 10,
=======
       "       tool strain    length flag position threshold\n",
       "0  sniffles   LP_J   6055203   TN        0      1000\n",
       "1  sniffles   LP_J  11015759   TN  6056928      1000\n",
       "2  sniffles   LP_J  11015759   TN  6056928      1000\n",
       "3  sniffles   LP_J  11015759   TN  6056928      1000\n",
       "4  sniffles   LP_J  11015759   TN  6056928      1000"
      ]
     },
     "execution_count": 9,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#1000 threshold non-deletions\n",
    "\n",
    "th=1000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_1000_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
<<<<<<< HEAD
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_1000_nondel = pd.concat([df_current, df_1000_nondel],ignore_index=True)\n",
    "            \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "\n",
    "        else:\n",
    "            print (\"Nonexistant File: \" + file)       \n",
    "df_1000_nondel.head()"
=======
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_1000_nondel = pd.concat([df_current, df_1000_nondel],ignore_index=True)\n",
    "            \n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "        \n",
    "df_1000_nondel.head()\n",
    "\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10000 bp Threshold"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:16: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  app.launch_new_instance()\n"
=======
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nonexistant File: ../raw_data/mouse/custom_vcf_fulldata/10000t/deletions/nf_10000t.pindel.BALB_CJ.chr19.100p_sorted.modified.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>34582728</td>\n",
       "      <td>17098523</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>48</td>\n",
       "      <td>17201451</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>34</td>\n",
       "      <td>17805736</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>37</td>\n",
       "      <td>18086146</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>47</td>\n",
       "      <td>18253183</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34</td>\n",
       "      <td>FP</td>\n",
       "      <td>17805736</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>37</td>\n",
       "      <td>FP</td>\n",
       "      <td>18086146</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>47</td>\n",
       "      <td>FP</td>\n",
       "      <td>18253183</td>\n",
       "      <td>10000</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length  position strain threshold      tool\n",
       "0   FP  34582728  17098523   LP_J     10000  sniffles\n",
       "1   FP        48  17201451   LP_J     10000  sniffles\n",
       "2   FP        34  17805736   LP_J     10000  sniffles\n",
       "3   FP        37  18086146   LP_J     10000  sniffles\n",
       "4   FP        47  18253183   LP_J     10000  sniffles"
      ]
     },
     "execution_count": 11,
=======
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J  34582728   FP  17098523     10000\n",
       "1  sniffles   LP_J        48   FP  17201451     10000\n",
       "2  sniffles   LP_J        34   FP  17805736     10000\n",
       "3  sniffles   LP_J        37   FP  18086146     10000\n",
       "4  sniffles   LP_J        47   FP  18253183     10000"
      ]
     },
     "execution_count": 10,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10000 threshold deletions\n",
    "#missing files: pindel.BALB_CJ, and platypus.BALB_CJ\n",
    "\n",
    "th=10000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10000 = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.vcf'\n",
    "        \n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_10000 = pd.concat([df_current, df_10000],ignore_index=True)\n",
    "            \n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "                \n",
    "        else:\n",
    "            print(\"Nonexistant File: \" + file)\n",
    "        \n",
    "df_10000.head()\n"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 12,
=======
   "execution_count": 11,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
<<<<<<< HEAD
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:15: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  from ipykernel import kernelapp as app\n"
=======
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty File: ../raw_data/mouse/custom_vcf_fulldata/10000t/non-deletions/nf_10000t.pindel.BALB_CJ.chr19.100p_sorted.modified.nondel.vcf\n"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>TN</td>\n",
       "      <td>6055203</td>\n",
       "      <td>0</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TN</td>\n",
       "      <td>11015759</td>\n",
       "      <td>6056928</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10000</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6055203</td>\n",
       "      <td>TN</td>\n",
       "      <td>0</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>11015759</td>\n",
       "      <td>TN</td>\n",
       "      <td>6056928</td>\n",
       "      <td>10000</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length position strain threshold      tool\n",
       "0   TN   6055203        0   LP_J     10000  sniffles\n",
       "1   TN  11015759  6056928   LP_J     10000  sniffles\n",
       "2   TN  11015759  6056928   LP_J     10000  sniffles\n",
       "3   TN  11015759  6056928   LP_J     10000  sniffles\n",
       "4   TN  11015759  6056928   LP_J     10000  sniffles"
      ]
     },
     "execution_count": 12,
=======
       "       tool strain    length flag position threshold\n",
       "0  sniffles   LP_J   6055203   TN        0     10000\n",
       "1  sniffles   LP_J  11015759   TN  6056928     10000\n",
       "2  sniffles   LP_J  11015759   TN  6056928     10000\n",
       "3  sniffles   LP_J  11015759   TN  6056928     10000\n",
       "4  sniffles   LP_J  11015759   TN  6056928     10000"
      ]
     },
     "execution_count": 11,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#10000 threshold non-deletions\n",
    "\n",
    "th=10000    \n",
    "df_current = pd.DataFrame(columns=['strain','length','flag','position','threshold'])\n",
    "df_10000_nondel = pd.DataFrame(columns=['tool','strain','length','flag','position','threshold'])\n",
    "\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        file='../raw_data/mouse/custom_vcf_fulldata/'+str(th)+'t/non-deletions/nf_'+str(th)+'t.'+t+'.'+s+'.chr19.100p_sorted.modified.nondel.vcf'\n",
<<<<<<< HEAD
    "\n",
    "        if os.path.exists(file):\n",
    "            callset = allel.read_vcf(file,fields='*')\n",
    "            if callset!=None:\n",
    "                df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "                df_10000_nondel = pd.concat([df_current, df_10000_nondel],ignore_index=True)\n",
    "\n",
    "            else:\n",
    "                print(\"Empty File: \" + file)\n",
    "\n",
    "        else:\n",
    "            print (\"Nonexistant File: \" + file)        \n",
=======
    "        \n",
    "        \n",
    "        callset = allel.read_vcf(file,fields='*')\n",
    "        if callset!=None:\n",
    "            df_current = pd.DataFrame({'tool': t, 'strain': s, 'length': callset['variants/SVLEN'],'flag': callset['variants/FLAG'],'position': callset['variants/POS'],'threshold': th})\n",
    "            df_10000_nondel = pd.concat([df_current, df_10000_nondel],ignore_index=True)\n",
    "        else:\n",
    "            print(\"Empty File: \" + file)\n",
    "           \n",
    "        \n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "df_10000_nondel.head()\n"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 13,
=======
   "execution_count": 12,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'df_10' (DataFrame)\n",
      "Stored 'df_100' (DataFrame)\n",
      "Stored 'df_1000' (DataFrame)\n",
      "Stored 'df_10000' (DataFrame)\n",
      "Stored 'df_10_nondel' (DataFrame)\n",
      "Stored 'df_100_nondel' (DataFrame)\n",
      "Stored 'df_1000_nondel' (DataFrame)\n",
      "Stored 'df_10000_nondel' (DataFrame)\n"
     ]
    }
   ],
   "source": [
    "%store df_10\n",
    "%store df_100\n",
    "%store df_1000\n",
    "%store df_10000\n",
    "%store df_10_nondel\n",
    "%store df_100_nondel\n",
    "%store df_1000_nondel\n",
    "%store df_10000_nondel"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 14,
=======
   "execution_count": 13,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [],
   "source": [
    "df_10.to_csv('../analysis_files/df_10.csv')\n",
    "df_100.to_csv('../analysis_files/df_100.csv')\n",
    "df_1000.to_csv('../analysis_files/df_1000.csv')\n",
    "df_10000.to_csv('../analysis_files/df_10000.csv')\n",
    "\n",
    "df_10_nondel.to_csv('../analysis_files/df_10_nondel.csv')\n",
    "df_100_nondel.to_csv('../analysis_files/df_100_nondel.csv')\n",
    "df_1000_nondel.to_csv('../analysis_files/df_1000_nondel.csv')\n",
    "df_10000_nondel.to_csv('../analysis_files/df_10000_nondel.csv')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 15,
=======
   "execution_count": 14,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [],
   "source": [
    "#List of your dataframes\n",
    "#removed df_0 and df_0_nondel for graphing purposes\n",
    "pdList = [df_10,df_100,df_1000,df_10000,df_10_nondel,df_100_nondel,df_1000_nondel,df_10000_nondel] \n",
    "df_cat = pd.concat(pdList)"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 16,
=======
   "execution_count": 15,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
<<<<<<< HEAD
       "      <th>flag</th>\n",
       "      <th>length</th>\n",
       "      <th>position</th>\n",
       "      <th>strain</th>\n",
       "      <th>threshold</th>\n",
       "      <th>tool</th>\n",
=======
       "      <th>tool</th>\n",
       "      <th>strain</th>\n",
       "      <th>length</th>\n",
       "      <th>flag</th>\n",
       "      <th>position</th>\n",
       "      <th>threshold</th>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
       "      <td>FP</td>\n",
       "      <td>1725</td>\n",
       "      <td>6055203</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FP</td>\n",
       "      <td>34582728</td>\n",
       "      <td>17098523</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FP</td>\n",
       "      <td>48</td>\n",
       "      <td>17201451</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FP</td>\n",
       "      <td>114</td>\n",
       "      <td>17231867</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FP</td>\n",
       "      <td>35</td>\n",
       "      <td>17291187</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>10</td>\n",
       "      <td>sniffles</td>\n",
=======
       "      <td>0</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>1725</td>\n",
       "      <td>FP</td>\n",
       "      <td>6055203</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>34582728</td>\n",
       "      <td>FP</td>\n",
       "      <td>17098523</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>48</td>\n",
       "      <td>FP</td>\n",
       "      <td>17201451</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>114</td>\n",
       "      <td>FP</td>\n",
       "      <td>17231867</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sniffles</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>35</td>\n",
       "      <td>FP</td>\n",
       "      <td>17291187</td>\n",
       "      <td>10</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
<<<<<<< HEAD
       "  flag    length  position strain threshold      tool\n",
       "0   FP      1725   6055203   LP_J        10  sniffles\n",
       "1   FP  34582728  17098523   LP_J        10  sniffles\n",
       "2   FP        48  17201451   LP_J        10  sniffles\n",
       "3   FP       114  17231867   LP_J        10  sniffles\n",
       "4   FP        35  17291187   LP_J        10  sniffles"
      ]
     },
     "execution_count": 16,
=======
       "       tool strain    length flag  position threshold\n",
       "0  sniffles   LP_J      1725   FP   6055203        10\n",
       "1  sniffles   LP_J  34582728   FP  17098523        10\n",
       "2  sniffles   LP_J        48   FP  17201451        10\n",
       "3  sniffles   LP_J       114   FP  17231867        10\n",
       "4  sniffles   LP_J        35   FP  17291187        10"
      ]
     },
     "execution_count": 15,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cat.head()"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 17,
=======
   "execution_count": 16,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
<<<<<<< HEAD
       "(357, 4)"
      ]
     },
     "execution_count": 17,
=======
       "(381, 4)"
      ]
     },
     "execution_count": 16,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#output counts for TP, FP, TN rows are not the same as Varuni's; due to missing files?\n",
    "group_data_TP=df_cat[df_cat['flag'] == 'TP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_TP=group_data_TP.rename(columns={\"flag\": \"nTP\"})\n",
<<<<<<< HEAD
    "group_data_TP.to_csv('../analysis_files/group_data_TP.csv')\n",
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "group_data_TP.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#group_data_TP.head(20)"
=======
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>A_J</td>\n",
       "      <td>304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>316</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         tool  threshold   strain  nTP\n",
       "0   BioGraph*         10    AKR_J   36\n",
       "1   BioGraph*         10      A_J   47\n",
       "2   BioGraph*         10  BALB_CJ   45\n",
       "3   BioGraph*         10  C3H_HeJ   41\n",
       "4   BioGraph*         10    CBA_J   44\n",
       "5   BioGraph*         10   DBA_2J   45\n",
       "6   BioGraph*         10     LP_J   39\n",
       "7   BioGraph*        100    AKR_J  249\n",
       "8   BioGraph*        100      A_J  279\n",
       "9   BioGraph*        100  BALB_CJ  269\n",
       "10  BioGraph*        100  C3H_HeJ  251\n",
       "11  BioGraph*        100    CBA_J  272\n",
       "12  BioGraph*        100   DBA_2J  285\n",
       "13  BioGraph*        100     LP_J  222\n",
       "14  BioGraph*       1000    AKR_J  269\n",
       "15  BioGraph*       1000      A_J  304\n",
       "16  BioGraph*       1000  BALB_CJ  294\n",
       "17  BioGraph*       1000  C3H_HeJ  278\n",
       "18  BioGraph*       1000    CBA_J  302\n",
       "19  BioGraph*       1000   DBA_2J  316"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TP.head(20)"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 19,
=======
   "execution_count": 18,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
<<<<<<< HEAD
       "(364, 4)"
      ]
     },
     "execution_count": 19,
=======
       "(388, 4)"
      ]
     },
     "execution_count": 18,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_FP=df_cat[df_cat['flag'] == 'FP'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_FP=group_data_FP.rename(columns={\"flag\": \"nFP\"})\n",
<<<<<<< HEAD
    "group_data_FP.to_csv('../analysis_files/group_data_FP.csv')\n",
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "group_data_FP.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 20,
=======
   "execution_count": 19,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nFP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
=======
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>1</th>\n",
=======
       "      <td>1</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>2</th>\n",
=======
       "      <td>2</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>3</th>\n",
=======
       "      <td>3</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>4</th>\n",
=======
       "      <td>4</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>445</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nFP\n",
       "0  BioGraph*         10    AKR_J  389\n",
       "1  BioGraph*         10      A_J  421\n",
       "2  BioGraph*         10  BALB_CJ  417\n",
       "3  BioGraph*         10  C3H_HeJ  416\n",
       "4  BioGraph*         10    CBA_J  445"
      ]
     },
<<<<<<< HEAD
     "execution_count": 20,
=======
     "execution_count": 19,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_FP.head()"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 21,
=======
   "execution_count": 20,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
<<<<<<< HEAD
       "(364, 4)"
      ]
     },
     "execution_count": 21,
=======
       "(388, 4)"
      ]
     },
     "execution_count": 20,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group_data_TN=df_cat[df_cat['flag'] == 'TN'].groupby(['tool','threshold','strain'],as_index=False)['flag'].count()\n",
    "group_data_TN=group_data_TN.rename(columns={\"flag\": \"nTN\"})\n",
<<<<<<< HEAD
    "group_data_TN.to_csv('../analysis_files/group_data_TN.csv')\n",
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "group_data_TN.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A_J gasv 10\n",
      "AKR_J gasv 10\n",
      "BALB_CJ gasv 10\n",
      "CBA_J gasv 10\n",
      "C3H_HeJ gasv 10\n",
      "DBA_2J gasv 10\n",
      "LP_J gasv 10\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(364, 4)"
      ]
     },
     "execution_count": 22,
=======
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(392, 4)"
      ]
     },
     "execution_count": 21,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#fill missing combinations with 0s\n",
    "#removed 0 threshold group since it = 0 for all tools\n",
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_TP['tool'] == t) & (group_data_TP['strain'] == s) & (group_data_TP['threshold'] == th)).any()):\n",
<<<<<<< HEAD
    "                print (s,t,th)\n",
=======
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "                group_data_TP = group_data_TP.append({'tool' : t , 'strain' : s,'threshold' : th,'nTP' : 0} , ignore_index=True)\n",
    "group_data_TP.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(364, 4)"
      ]
     },
     "execution_count": 23,
=======
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BALB_CJ pindel 10\n",
      "BALB_CJ pindel 100\n",
      "BALB_CJ pindel 1000\n",
      "BALB_CJ pindel 10000\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(388, 4)"
      ]
     },
     "execution_count": 22,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_FP['tool'] == t) & (group_data_FP['strain'] == s) & (group_data_FP['threshold'] == th)).any()):\n",
    "                #group_data_FP = group_data_FP.append({'tool' : t , 'strain' : s,'threshold' : th,'nFP' : 0} , ignore_index=True)\n",
    "                print (s,t,th)\n",
    "group_data_FP.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 24,
=======
   "execution_count": 23,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
<<<<<<< HEAD
       "(364, 4)"
      ]
     },
     "execution_count": 24,
=======
       "(392, 4)"
      ]
     },
     "execution_count": 23,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for s in strain:\n",
    "    for t in tools:\n",
    "        for th in [10,100,1000,10000]:\n",
    "            if not (((group_data_TN['tool'] == t) & (group_data_TN['strain'] == s) & (group_data_TN['threshold'] == th)).any()):\n",
    "                group_data_TN = group_data_TN.append({'tool' : t , 'strain' : s,'threshold' : th,'nTN' : 0} , ignore_index=True)\n",
<<<<<<< HEAD
    "                print (s,t,th)\n",
=======
    "                #print (s,t,th)\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
    "group_data_TN.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 25,
=======
   "execution_count": 24,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
<<<<<<< HEAD
       "(364, 6)"
      ]
     },
     "execution_count": 25,
=======
       "(388, 6)"
      ]
     },
     "execution_count": 24,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge TP and FP\n",
    "df_merge=pd.merge(group_data_TP, group_data_FP,on=['tool','threshold','strain'])\n",
    "df_merge=pd.merge(df_merge, group_data_TN,on=['tool','threshold','strain'])\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 26,
=======
   "execution_count": 25,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
=======
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>1</th>\n",
=======
       "      <td>1</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>A_J</td>\n",
       "      <td>47</td>\n",
       "      <td>421</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>2</th>\n",
=======
       "      <td>2</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>45</td>\n",
       "      <td>417</td>\n",
       "      <td>237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>3</th>\n",
=======
       "      <td>3</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>41</td>\n",
       "      <td>416</td>\n",
       "      <td>247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>4</th>\n",
=======
       "      <td>4</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>44</td>\n",
       "      <td>445</td>\n",
       "      <td>270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>5</th>\n",
=======
       "      <td>5</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>45</td>\n",
       "      <td>454</td>\n",
       "      <td>279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>6</th>\n",
=======
       "      <td>6</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>39</td>\n",
       "      <td>336</td>\n",
       "      <td>213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>7</th>\n",
=======
       "      <td>7</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>8</th>\n",
=======
       "      <td>8</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>A_J</td>\n",
       "      <td>279</td>\n",
       "      <td>189</td>\n",
       "      <td>317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>9</th>\n",
=======
       "      <td>9</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>269</td>\n",
       "      <td>193</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        tool  threshold   strain  nTP  nFP  nTN\n",
       "0  BioGraph*         10    AKR_J   36  389  206\n",
       "1  BioGraph*         10      A_J   47  421  220\n",
       "2  BioGraph*         10  BALB_CJ   45  417  237\n",
       "3  BioGraph*         10  C3H_HeJ   41  416  247\n",
       "4  BioGraph*         10    CBA_J   44  445  270\n",
       "5  BioGraph*         10   DBA_2J   45  454  279\n",
       "6  BioGraph*         10     LP_J   39  336  213\n",
       "7  BioGraph*        100    AKR_J  249  176  289\n",
       "8  BioGraph*        100      A_J  279  189  317\n",
       "9  BioGraph*        100  BALB_CJ  269  193  330"
      ]
     },
<<<<<<< HEAD
     "execution_count": 26,
=======
     "execution_count": 25,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.head(10)"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vs/.local/lib/python3.5/site-packages/ipykernel_launcher.py:11: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKR_J</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A_J</td>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBA_J</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LP_J</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    strain  n_true\n",
       "0    AKR_J     504\n",
       "1      A_J     533\n",
       "2  BALB_CJ     545\n",
       "3  C3H_HeJ     539\n",
       "4    CBA_J     586\n",
       "5   DBA_2J     609\n",
       "6     LP_J     483"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#true SV deletions\n",
    "df_current = pd.DataFrame(columns=['strain','length'])\n",
    "df = pd.DataFrame(columns=['strain','length'])\n",
    "\n",
    "\n",
    "for s in strain:\n",
    "    file='../gold_standard/mouse_vcf/'+s+'_reference.vcf'\n",
    "    callset = allel.read_vcf(file,fields='*')\n",
    "    \n",
    "    df_current = pd.DataFrame({'strain': s, 'length': callset['variants/SVLEN']})\n",
    "    df = pd.concat([df_current, df],ignore_index=True)\n",
    "group_data_true = df.groupby(['strain'],as_index=False).count()\n",
    "group_data_true=group_data_true.rename(columns={\"length\": \"n_true\"})\n",
    "group_data_true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(364, 7)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge=pd.merge(df_merge, group_data_true)\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
=======
   "execution_count": 26,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
<<<<<<< HEAD
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>9</td>\n",
       "      <td>527</td>\n",
       "      <td>253</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>254</td>\n",
       "      <td>282</td>\n",
       "      <td>314</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>311</td>\n",
       "      <td>225</td>\n",
       "      <td>348</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>325</td>\n",
       "      <td>211</td>\n",
       "      <td>393</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>clever</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>54</td>\n",
       "      <td>9551</td>\n",
       "      <td>31</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>clever</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>336</td>\n",
       "      <td>9269</td>\n",
       "      <td>53</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>clever</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>396</td>\n",
       "      <td>9209</td>\n",
       "      <td>75</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>clever</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>488</td>\n",
       "      <td>9117</td>\n",
       "      <td>632</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>delly</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>69</td>\n",
       "      <td>854</td>\n",
       "      <td>117</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>delly</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>311</td>\n",
       "      <td>612</td>\n",
       "      <td>188</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>delly</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>353</td>\n",
       "      <td>570</td>\n",
       "      <td>208</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>delly</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>405</td>\n",
       "      <td>518</td>\n",
       "      <td>379</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>gasv</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>68</td>\n",
       "      <td>12328</td>\n",
       "      <td>9</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>gasv</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>304</td>\n",
       "      <td>12092</td>\n",
       "      <td>35</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>gasv</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>496</td>\n",
       "      <td>11900</td>\n",
       "      <td>508</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>gridss</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>58</td>\n",
       "      <td>2376</td>\n",
       "      <td>46</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>gridss</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>317</td>\n",
       "      <td>2117</td>\n",
       "      <td>84</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>gridss</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>384</td>\n",
       "      <td>2050</td>\n",
       "      <td>123</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>gridss</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>464</td>\n",
       "      <td>1970</td>\n",
       "      <td>500</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>14</td>\n",
       "      <td>12267</td>\n",
       "      <td>201</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>82</td>\n",
       "      <td>12199</td>\n",
       "      <td>201</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>181</td>\n",
       "      <td>12100</td>\n",
       "      <td>227</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>305</td>\n",
       "      <td>11976</td>\n",
       "      <td>1065</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>17</td>\n",
       "      <td>6534</td>\n",
       "      <td>53</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>110</td>\n",
       "      <td>6441</td>\n",
       "      <td>57</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>195</td>\n",
       "      <td>6356</td>\n",
       "      <td>70</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334</th>\n",
       "      <td>gridss</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>446</td>\n",
       "      <td>1889</td>\n",
       "      <td>560</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>15</td>\n",
       "      <td>13401</td>\n",
       "      <td>137</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>67</td>\n",
       "      <td>13349</td>\n",
       "      <td>137</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>189</td>\n",
       "      <td>13227</td>\n",
       "      <td>159</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>indelminer</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>347</td>\n",
       "      <td>13069</td>\n",
       "      <td>1265</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>18</td>\n",
       "      <td>5554</td>\n",
       "      <td>54</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>97</td>\n",
       "      <td>5475</td>\n",
       "      <td>59</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>183</td>\n",
       "      <td>5389</td>\n",
       "      <td>82</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>mistrvar</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>430</td>\n",
       "      <td>5142</td>\n",
       "      <td>789</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>pindel</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>43</td>\n",
       "      <td>76321</td>\n",
       "      <td>43</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>pindel</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>195</td>\n",
       "      <td>76169</td>\n",
       "      <td>45</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>pindel</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>340</td>\n",
       "      <td>76024</td>\n",
       "      <td>131</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346</th>\n",
       "      <td>pindel</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>480</td>\n",
       "      <td>75884</td>\n",
       "      <td>4239</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>popdel</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>6</td>\n",
       "      <td>216</td>\n",
       "      <td>256</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>348</th>\n",
       "      <td>popdel</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>173</td>\n",
       "      <td>49</td>\n",
       "      <td>317</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>349</th>\n",
       "      <td>popdel</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>210</td>\n",
       "      <td>12</td>\n",
       "      <td>336</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>popdel</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>215</td>\n",
       "      <td>7</td>\n",
       "      <td>359</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>351</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>4</td>\n",
       "      <td>2662</td>\n",
       "      <td>53</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>132</td>\n",
       "      <td>2534</td>\n",
       "      <td>27</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>227</td>\n",
       "      <td>2439</td>\n",
       "      <td>59</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>rdxplorer</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>393</td>\n",
       "      <td>2273</td>\n",
       "      <td>187</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>smoove</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>78</td>\n",
       "      <td>322</td>\n",
       "      <td>245</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>smoove</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>281</td>\n",
       "      <td>119</td>\n",
       "      <td>331</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>smoove</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>315</td>\n",
       "      <td>85</td>\n",
       "      <td>348</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>smoove</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>331</td>\n",
       "      <td>69</td>\n",
       "      <td>423</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
=======
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strain</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A_J</td>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BALB_CJ</td>\n",
       "      <td>545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C3H_HeJ</td>\n",
       "      <td>539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>CBA_J</td>\n",
       "      <td>586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>DBA_2J</td>\n",
       "      <td>609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    strain  n_true\n",
       "0    AKR_J     504\n",
       "1      A_J     533\n",
       "2  BALB_CJ     545\n",
       "3  C3H_HeJ     539\n",
       "4    CBA_J     586\n",
       "5   DBA_2J     609\n",
       "6     LP_J     483"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#true SV deletions\n",
    "df_current = pd.DataFrame(columns=['strain','length'])\n",
    "df = pd.DataFrame(columns=['strain','length'])\n",
    "\n",
    "\n",
    "for s in strain:\n",
    "    file='../gold_standard/mouse_vcf/'+s+'_reference.vcf'\n",
    "    callset = allel.read_vcf(file,fields='*')\n",
    "    \n",
    "    df_current = pd.DataFrame({'strain': s, 'length': callset['variants/SVLEN']})\n",
    "    df = pd.concat([df_current, df],ignore_index=True)\n",
    "group_data_true = df.groupby(['strain'],as_index=False).count()\n",
    "group_data_true=group_data_true.rename(columns={\"length\": \"n_true\"})\n",
    "group_data_true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(388, 7)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge=pd.merge(df_merge, group_data_true)\n",
    "df_merge.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>9</td>\n",
       "      <td>527</td>\n",
       "      <td>253</td>\n",
       "      <td>504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>383</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>sniffles</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>44</td>\n",
       "      <td>142</td>\n",
       "      <td>352</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>360</th>\n",
=======
       "      <td>384</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>sniffles</td>\n",
       "      <td>100</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>121</td>\n",
       "      <td>65</td>\n",
       "      <td>366</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>361</th>\n",
=======
       "      <td>385</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>sniffles</td>\n",
       "      <td>1000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>129</td>\n",
       "      <td>57</td>\n",
       "      <td>369</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>362</th>\n",
=======
       "      <td>386</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>sniffles</td>\n",
       "      <td>10000</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>143</td>\n",
       "      <td>43</td>\n",
       "      <td>382</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>363</th>\n",
=======
       "      <td>387</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>gasv</td>\n",
       "      <td>10</td>\n",
       "      <td>LP_J</td>\n",
       "      <td>0</td>\n",
       "      <td>11605</td>\n",
       "      <td>9</td>\n",
       "      <td>483</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
<<<<<<< HEAD
       "<p>364 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            tool  threshold strain  nTP    nFP   nTN  n_true\n",
       "0      BioGraph*         10  AKR_J   36    389   206     504\n",
       "1      BioGraph*        100  AKR_J  249    176   289     504\n",
       "2      BioGraph*       1000  AKR_J  269    156   299     504\n",
       "3      BioGraph*      10000  AKR_J  291    134   347     504\n",
       "4    breakdancer         10  AKR_J    9    527   253     504\n",
       "5    breakdancer        100  AKR_J  254    282   314     504\n",
       "6    breakdancer       1000  AKR_J  311    225   348     504\n",
       "7    breakdancer      10000  AKR_J  325    211   393     504\n",
       "8         clever         10  AKR_J   54   9551    31     504\n",
       "9         clever        100  AKR_J  336   9269    53     504\n",
       "10        clever       1000  AKR_J  396   9209    75     504\n",
       "11        clever      10000  AKR_J  488   9117   632     504\n",
       "12         delly         10  AKR_J   69    854   117     504\n",
       "13         delly        100  AKR_J  311    612   188     504\n",
       "14         delly       1000  AKR_J  353    570   208     504\n",
       "15         delly      10000  AKR_J  405    518   379     504\n",
       "16          gasv        100  AKR_J   68  12328     9     504\n",
       "17          gasv       1000  AKR_J  304  12092    35     504\n",
       "18          gasv      10000  AKR_J  496  11900   508     504\n",
       "19        gridss         10  AKR_J   58   2376    46     504\n",
       "20        gridss        100  AKR_J  317   2117    84     504\n",
       "21        gridss       1000  AKR_J  384   2050   123     504\n",
       "22        gridss      10000  AKR_J  464   1970   500     504\n",
       "23    indelminer         10  AKR_J   14  12267   201     504\n",
       "24    indelminer        100  AKR_J   82  12199   201     504\n",
       "25    indelminer       1000  AKR_J  181  12100   227     504\n",
       "26    indelminer      10000  AKR_J  305  11976  1065     504\n",
       "27      mistrvar         10  AKR_J   17   6534    53     504\n",
       "28      mistrvar        100  AKR_J  110   6441    57     504\n",
       "29      mistrvar       1000  AKR_J  195   6356    70     504\n",
       "..           ...        ...    ...  ...    ...   ...     ...\n",
       "334       gridss      10000   LP_J  446   1889   560     483\n",
       "335   indelminer         10   LP_J   15  13401   137     483\n",
       "336   indelminer        100   LP_J   67  13349   137     483\n",
       "337   indelminer       1000   LP_J  189  13227   159     483\n",
       "338   indelminer      10000   LP_J  347  13069  1265     483\n",
       "339     mistrvar         10   LP_J   18   5554    54     483\n",
       "340     mistrvar        100   LP_J   97   5475    59     483\n",
       "341     mistrvar       1000   LP_J  183   5389    82     483\n",
       "342     mistrvar      10000   LP_J  430   5142   789     483\n",
       "343       pindel         10   LP_J   43  76321    43     483\n",
       "344       pindel        100   LP_J  195  76169    45     483\n",
       "345       pindel       1000   LP_J  340  76024   131     483\n",
       "346       pindel      10000   LP_J  480  75884  4239     483\n",
       "347       popdel         10   LP_J    6    216   256     483\n",
       "348       popdel        100   LP_J  173     49   317     483\n",
       "349       popdel       1000   LP_J  210     12   336     483\n",
       "350       popdel      10000   LP_J  215      7   359     483\n",
       "351    rdxplorer         10   LP_J    4   2662    53     483\n",
       "352    rdxplorer        100   LP_J  132   2534    27     483\n",
       "353    rdxplorer       1000   LP_J  227   2439    59     483\n",
       "354    rdxplorer      10000   LP_J  393   2273   187     483\n",
       "355       smoove         10   LP_J   78    322   245     483\n",
       "356       smoove        100   LP_J  281    119   331     483\n",
       "357       smoove       1000   LP_J  315     85   348     483\n",
       "358       smoove      10000   LP_J  331     69   423     483\n",
       "359     sniffles         10   LP_J   44    142   352     483\n",
       "360     sniffles        100   LP_J  121     65   366     483\n",
       "361     sniffles       1000   LP_J  129     57   369     483\n",
       "362     sniffles      10000   LP_J  143     43   382     483\n",
       "363         gasv         10   LP_J    0  11605     9     483\n",
       "\n",
       "[364 rows x 7 columns]"
      ]
     },
     "execution_count": 29,
=======
       "<p>388 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            tool  threshold strain  nTP    nFP  nTN  n_true\n",
       "0      BioGraph*         10  AKR_J   36    389  206     504\n",
       "1      BioGraph*        100  AKR_J  249    176  289     504\n",
       "2      BioGraph*       1000  AKR_J  269    156  299     504\n",
       "3      BioGraph*      10000  AKR_J  291    134  347     504\n",
       "4    breakdancer         10  AKR_J    9    527  253     504\n",
       "..           ...        ...    ...  ...    ...  ...     ...\n",
       "383     sniffles         10   LP_J   44    142  352     483\n",
       "384     sniffles        100   LP_J  121     65  366     483\n",
       "385     sniffles       1000   LP_J  129     57  369     483\n",
       "386     sniffles      10000   LP_J  143     43  382     483\n",
       "387         gasv         10   LP_J    0  11605    9     483\n",
       "\n",
       "[388 rows x 7 columns]"
      ]
     },
     "execution_count": 28,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating precision, sensitivity, specificity, f-score"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 30,
=======
   "execution_count": 29,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "      <th>nFN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
=======
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "      <td>468</td>\n",
       "      <td>0.071429</td>\n",
       "      <td>0.084706</td>\n",
       "      <td>0.346218</td>\n",
       "      <td>0.077503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>1</th>\n",
=======
       "      <td>1</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "      <td>255</td>\n",
       "      <td>0.494048</td>\n",
       "      <td>0.585882</td>\n",
       "      <td>0.621505</td>\n",
       "      <td>0.536060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>2</th>\n",
=======
       "      <td>2</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "      <td>235</td>\n",
       "      <td>0.533730</td>\n",
       "      <td>0.632941</td>\n",
       "      <td>0.657143</td>\n",
       "      <td>0.579117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>3</th>\n",
=======
       "      <td>3</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "      <td>213</td>\n",
       "      <td>0.577381</td>\n",
       "      <td>0.684706</td>\n",
       "      <td>0.721414</td>\n",
       "      <td>0.626480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>4</th>\n",
=======
       "      <td>4</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>9</td>\n",
       "      <td>527</td>\n",
       "      <td>253</td>\n",
       "      <td>504</td>\n",
       "      <td>495</td>\n",
       "      <td>0.017857</td>\n",
       "      <td>0.016791</td>\n",
       "      <td>0.324359</td>\n",
       "      <td>0.017308</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold strain  nTP  nFP  nTN  n_true  nFN  sensitivity  \\\n",
       "0    BioGraph*         10  AKR_J   36  389  206     504  468     0.071429   \n",
       "1    BioGraph*        100  AKR_J  249  176  289     504  255     0.494048   \n",
       "2    BioGraph*       1000  AKR_J  269  156  299     504  235     0.533730   \n",
       "3    BioGraph*      10000  AKR_J  291  134  347     504  213     0.577381   \n",
       "4  breakdancer         10  AKR_J    9  527  253     504  495     0.017857   \n",
       "\n",
       "   precision  specificity   f-score  \n",
       "0   0.084706     0.346218  0.077503  \n",
       "1   0.585882     0.621505  0.536060  \n",
       "2   0.632941     0.657143  0.579117  \n",
       "3   0.684706     0.721414  0.626480  \n",
       "4   0.016791     0.324359  0.017308  "
      ]
     },
<<<<<<< HEAD
     "execution_count": 30,
=======
     "execution_count": 29,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge['nFN']=df_merge['n_true']-df_merge['nTP']\n",
    "\n",
    "df_merge['sensitivity']=df_merge['nTP']/df_merge['n_true']\n",
    "df_merge['precision']=df_merge['nTP']/(df_merge['nTP']+df_merge['nFP'])\n",
    "df_merge['specificity']=df_merge['nTN']/(df_merge['nTN']+df_merge['nFP'])\n",
    "df_merge['f-score']=2*(df_merge['sensitivity']*df_merge['precision'])/(df_merge['sensitivity']+df_merge['precision']+0.00000001)\n",
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 31,
=======
   "execution_count": 30,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nFN</th>\n",
       "      <th>nTN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
=======
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>297</td>\n",
       "      <td>2878</td>\n",
       "      <td>3502</td>\n",
       "      <td>1672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>1</th>\n",
=======
       "      <td>1</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>1827</td>\n",
       "      <td>1348</td>\n",
       "      <td>1972</td>\n",
       "      <td>2249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>2</th>\n",
=======
       "      <td>2</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>2004</td>\n",
       "      <td>1171</td>\n",
       "      <td>1795</td>\n",
       "      <td>2331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>3</th>\n",
=======
       "      <td>3</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>2163</td>\n",
       "      <td>1012</td>\n",
       "      <td>1636</td>\n",
       "      <td>2717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>4</th>\n",
=======
       "      <td>4</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>58</td>\n",
       "      <td>3484</td>\n",
       "      <td>3741</td>\n",
       "      <td>2026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>5</th>\n",
=======
       "      <td>5</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>breakdancer</td>\n",
       "      <td>100</td>\n",
       "      <td>1883</td>\n",
       "      <td>1659</td>\n",
       "      <td>1916</td>\n",
       "      <td>2483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>6</th>\n",
=======
       "      <td>6</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>breakdancer</td>\n",
       "      <td>1000</td>\n",
       "      <td>2420</td>\n",
       "      <td>1122</td>\n",
       "      <td>1379</td>\n",
       "      <td>2813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>7</th>\n",
=======
       "      <td>7</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>breakdancer</td>\n",
       "      <td>10000</td>\n",
       "      <td>2522</td>\n",
       "      <td>1020</td>\n",
       "      <td>1277</td>\n",
       "      <td>3247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>8</th>\n",
=======
       "      <td>8</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>clever</td>\n",
       "      <td>10</td>\n",
       "      <td>486</td>\n",
       "      <td>13270</td>\n",
       "      <td>3313</td>\n",
       "      <td>1086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>9</th>\n",
=======
       "      <td>9</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>clever</td>\n",
       "      <td>100</td>\n",
       "      <td>2613</td>\n",
       "      <td>11143</td>\n",
       "      <td>1186</td>\n",
       "      <td>1963</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold   nTP    nFP   nFN   nTN\n",
       "0    BioGraph*         10   297   2878  3502  1672\n",
       "1    BioGraph*        100  1827   1348  1972  2249\n",
       "2    BioGraph*       1000  2004   1171  1795  2331\n",
       "3    BioGraph*      10000  2163   1012  1636  2717\n",
       "4  breakdancer         10    58   3484  3741  2026\n",
       "5  breakdancer        100  1883   1659  1916  2483\n",
       "6  breakdancer       1000  2420   1122  1379  2813\n",
       "7  breakdancer      10000  2522   1020  1277  3247\n",
       "8       clever         10   486  13270  3313  1086\n",
       "9       clever        100  2613  11143  1186  1963"
      ]
     },
<<<<<<< HEAD
     "execution_count": 31,
=======
     "execution_count": 30,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#count number of TP/FP/TN/FN for each tool at each threshold\n",
    "df_sum=df_merge.groupby(['tool','threshold'],as_index=False)['nTP','nFP','nFN','nTN'].sum()\n",
    "df_sum.head(10)"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 32,
=======
   "execution_count": 31,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stored 'df_sum' (DataFrame)\n"
     ]
    }
   ],
   "source": [
    "%store df_sum"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sum.to_csv('../analysis_files/df_sum.csv')"
=======
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sum.to_csv('../analysis_files/df_sum')"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 34,
=======
   "execution_count": 33,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [],
   "source": [
    "df_merge.to_csv('df_merged.csv')"
   ]
  },
  {
   "cell_type": "code",
<<<<<<< HEAD
   "execution_count": 35,
=======
   "execution_count": 34,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tool</th>\n",
       "      <th>threshold</th>\n",
       "      <th>strain</th>\n",
       "      <th>nTP</th>\n",
       "      <th>nFP</th>\n",
       "      <th>nTN</th>\n",
       "      <th>n_true</th>\n",
       "      <th>nFN</th>\n",
       "      <th>sensitivity</th>\n",
       "      <th>precision</th>\n",
       "      <th>specificity</th>\n",
       "      <th>f-score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>0</th>\n",
=======
       "      <td>0</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>36</td>\n",
       "      <td>389</td>\n",
       "      <td>206</td>\n",
       "      <td>504</td>\n",
       "      <td>468</td>\n",
       "      <td>0.071429</td>\n",
       "      <td>0.084706</td>\n",
       "      <td>0.346218</td>\n",
       "      <td>0.077503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>1</th>\n",
=======
       "      <td>1</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>100</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>249</td>\n",
       "      <td>176</td>\n",
       "      <td>289</td>\n",
       "      <td>504</td>\n",
       "      <td>255</td>\n",
       "      <td>0.494048</td>\n",
       "      <td>0.585882</td>\n",
       "      <td>0.621505</td>\n",
       "      <td>0.536060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>2</th>\n",
=======
       "      <td>2</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>1000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>269</td>\n",
       "      <td>156</td>\n",
       "      <td>299</td>\n",
       "      <td>504</td>\n",
       "      <td>235</td>\n",
       "      <td>0.533730</td>\n",
       "      <td>0.632941</td>\n",
       "      <td>0.657143</td>\n",
       "      <td>0.579117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>3</th>\n",
=======
       "      <td>3</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>BioGraph*</td>\n",
       "      <td>10000</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>291</td>\n",
       "      <td>134</td>\n",
       "      <td>347</td>\n",
       "      <td>504</td>\n",
       "      <td>213</td>\n",
       "      <td>0.577381</td>\n",
       "      <td>0.684706</td>\n",
       "      <td>0.721414</td>\n",
       "      <td>0.626480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
<<<<<<< HEAD
       "      <th>4</th>\n",
=======
       "      <td>4</td>\n",
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
       "      <td>breakdancer</td>\n",
       "      <td>10</td>\n",
       "      <td>AKR_J</td>\n",
       "      <td>9</td>\n",
       "      <td>527</td>\n",
       "      <td>253</td>\n",
       "      <td>504</td>\n",
       "      <td>495</td>\n",
       "      <td>0.017857</td>\n",
       "      <td>0.016791</td>\n",
       "      <td>0.324359</td>\n",
       "      <td>0.017308</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          tool  threshold strain  nTP  nFP  nTN  n_true  nFN  sensitivity  \\\n",
       "0    BioGraph*         10  AKR_J   36  389  206     504  468     0.071429   \n",
       "1    BioGraph*        100  AKR_J  249  176  289     504  255     0.494048   \n",
       "2    BioGraph*       1000  AKR_J  269  156  299     504  235     0.533730   \n",
       "3    BioGraph*      10000  AKR_J  291  134  347     504  213     0.577381   \n",
       "4  breakdancer         10  AKR_J    9  527  253     504  495     0.017857   \n",
       "\n",
       "   precision  specificity   f-score  \n",
       "0   0.084706     0.346218  0.077503  \n",
       "1   0.585882     0.621505  0.536060  \n",
       "2   0.632941     0.657143  0.579117  \n",
       "3   0.684706     0.721414  0.626480  \n",
       "4   0.016791     0.324359  0.017308  "
      ]
     },
<<<<<<< HEAD
     "execution_count": 35,
=======
     "execution_count": 34,
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
<<<<<<< HEAD
   "version": "3.5.2"
=======
   "version": "3.7.4"
>>>>>>> c1713ec61b27b47b24a432cc080a8565b90a9405
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
